cpp/src/gandiva/arrow.h

```diff
@@ -25,6 +25,7 @@
 #include <arrow/builder.h>
 #include <arrow/pretty_print.h>
 #include <arrow/record_batch.h>
+#include <arrow/result.h>
```

Comment:
Can't we use `type_fwd.h` instead?

Reply:
I tried removing the others and minimizing but it unfolded in too many changes.